### PR TITLE
[SYCLomatic] Refactor infrastructure to run analysis pass at first before migration

### PR DESCRIPTION
**Pull request by ziranzha**
Signed-off-by: Ziran Zhang <ziran.zhang@intel.com>

### CODE REVIEW

File: clang/test/dpct/d_dh_constant_db.cpp [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// UNSUPPORTED: -linux-
```
Comment (zhimingwang36) [inline clang/test/dpct/d_dh_constant_db.cpp:1]: Ask comment from zhiwei on comment as here disable the test case.

File: clang/lib/DPCT/Checkpoint.h [hunk 2]
```diff
@@ -57,7 +57,7 @@ enum {
 
 class AstCPStageMaintainer {
 public:
-  ~AstCPStageMaintainer() { CheckPointStage = CHECKPOINT_PROCESSING_FILE; }
+  ~AstCPStageMaintainer() { CheckPointStage = CHECKPOINT_UNKNOWN; }
```
Comment (zhimingwang36) [inline clang/lib/DPCT/Checkpoint.h]: we need refine the Checkpoint status as you include extra status.

File: clang/lib/DPCT/MigrationAction.cpp [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    CHECKPOINT_ReplacementPostProcess_EXIT();
+  }
+}
+
+void DpctToolAction::runPasses() {
+  for (auto Pass : Passes) {
+    runPass(Pass);
+
+    // CHECKPOINT_ReplacementPostProcess_ENTRY(RetJmp);
+    // if (RetJmp == 0) {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/MigrationAction.cpp]: remove the code commented out already.

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 4]
```diff
@@ -5678,9 +5665,9 @@ void BLASFunctionCallRule::runRule(const MatchFinder::MatchResult &Result) {
       CE->getDirectCallee()->getNameInfo().getName().getAsString();
 
   const SourceManager *SM = Result.SourceManager;
-  auto SL = SM->getExpansionLoc(CE->getBeginLoc());
-  std::string Key = SM->getFilename(SL).str() +
-                    std::to_string(SM->getDecomposedLoc(SL).second);
+  // auto SL = SM->getExpansionLoc(CE->getBeginLoc());
```
Comment (zhiweij1) [inline clang/lib/DPCT/ASTTraversal.cpp]: Debug code

**Review by zhiweij1 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Comment by zhimingwang36** [comment]
PRE test shows PRE-CI has passed,  so merged directly.

**Pull request merged**